### ISSUE TITLE
CocoaPods Deploy is failing 

### ISSUE THREAD

**Issue by winsmith** [TelemetryDeck/SwiftSDK#155]

**Comment by kkostov** [comment]
Hi @winsmith,

While working on https://github.com/TelemetryDeck/FlutterSDK/issues/11 I realized the latest [published version](https://cocoapods.org/pods/TelemetryClient) of the TelemetryClient is `1.5.1` and it doesn't include the needed changes from #161.

Do you have an update about the status of this issue? Is it something I can have a look at?

**Comment by Jeehut** [comment]
Just seeing this (the GitHub inbox seems to be broken for me). All I can say is that last time I checked this seemed to have been an authentication issue, and I don't have access to the email which authenticates on CocoaPods trunk, so I couldn't fix it. If you have accessed, please go ahead and fix if you can.

**Comment by kkostov** [comment]
@Jeehut, I checked the failing build log, it seems to be related to a build error during the validation step: 

```
    - NOTE  | xcodebuild:  note: Metadata extraction skipped. No AppIntents.framework dependency found. (in target 'App' from project 'App')
    - NOTE  | xcodebuild:  note: Using codesigning identity override: 
    - ERROR | [visionOS] unknown: Encountered an unknown error (Could not find a `xros` simulator (valid values: ios, tvos, watchos). Ensure that Xcode -> Window -> Devices has at least one `xros` simulator listed or otherwise add one.) during validation.
```

https://github.com/TelemetryDeck/SwiftSDK/actions/runs/9817450273/job/27108828828

I also don't have access to publish new versions, but perhaps removing "xros" would fix the build and it will work?

CocoaPods should not be deprecated though, it's also the only way to get dependencies in Flutter libraries at the moment. So please, do continue to maintain it 😄.

**Comment by Jeehut** [comment]
@kkostov Don't get me wrong, I'm not saying we should deprecate CocoaPods support. All I was trying to say is from a Swift developers perspective, CocoaPods should be seen as a dependency manager to move away from and to adopt SwiftPM instead. There's no reason for iOS developers to stay with CocoaPods, all the initial reasons like missing binary support have been fixed long time ago.

But I didn't know the Flutter support only worked through CocoaPods. I wonder why there's no SwiftPM-based way. But that's a separate topic, I guess. If the reality is that CocoaPods is needed for Flutter today, let's fix it.

The build log looks like the issue actually lies within the CI environment. It seems that is does not have a current Xcode version with a visionOS simulator in it. A fix could be as simple as updating the environment (if an updated one is available), or adding a step that explicitly installs the simulator.

Having that said, in version 2.0.0 I had not added the visionOS requirement for CocoaPods and IIRC then the reason it was failing was an authentication issue. So feel free to tackle the CI issue if you want it fixed fast. I can take a look later today or tomorrow if you don't find a solution. Feel free to ping me on Slack.

**Comment by kkostov** [comment]
@Jeehut it seems like you've already analysed the issue in a deeper level, so feel free to push a fix! Once published, I can wrap up https://github.com/TelemetryDeck/FlutterSDK/issues/11 so the sooner, the better!

**Comment by Jeehut** [comment]
@kkostov I just fixed the issue with the visionOS Simulator not being recognized in bc7e2a477b021e6d8918524fb360665e794d5d28 by making sure the CI uses the lastest Xcode version. But like I had mentioned above, there is an authentication issue with CocoaPods that I can't fix myself as I don't have access to the email the library was registered on CocoaPods. I guess only @winsmith can fix this by updating the `COCOAPODS_TRUNK_TOKEN` secret with a valid token. Or by adding me as an owner on CocoaPods, see [the docs](https://guides.cocoapods.org/making/getting-setup-with-trunk#adding-other-people-as-contributors).

Having this said, I want to also bring up that the podspec file specifies a library named "TelemetryClient" which we renamed to "TelemetryDeck". I'm not sure if CocoaPods has a renaming feature, so we might be better off creating a new CocoaPods library in the first place. The old one would continue to work for people who are on the old version. But upgrading to 2.0 would require a rename in their Podfile. We should also update our docs if we decide to rename and create a new lib. I can do both the creation of the new lib and the docs update if we decide to go this route. I would then invite @winsmith as another owner on the CocoaPods end.

By the way: Weirdly enough, when I go to cocoapods.org and enter "TelemetryClient" to search for the lib, there are no results. I wonder why that is…

**Comment by Jeehut** [comment]
@kkostov Together with Daniel we decided to make a new pod with a new name – and I just released it, you can find it here:
https://cocoapods.org/pods/TelemetryDeck

Note that the new name of the pod is `TelemetryDeck` and the first available version is `2.2.3`. This should hopefully lift off the blocking dependency for the Flutter SDK. Let me know if anything still doesn't work.

As for the title of this issue, stating the the Github Actions CI _deployment_ failed: In fact, I made the `2.2.3` release on my local machine because I wanted to make sure I have access to CocoaPods Trunk so I can (re)generate an access token (thus I cancelled the [deploy task](https://github.com/TelemetryDeck/SwiftSDK/actions/runs/9940551995)). But I did generate a new token and @winsmith updated the secret, so the authentication issue _should_ be fixed now. (We'll know upon next release!)

Next, I'm gonna update the CocoaPods setup docs to point to the new pod. Once that PR is merged, I think this can be closed for good. We can reopen if the next version update has any deployment issues again.

**Comment by kkostov** [comment]
Thanks @Jeehut and @winsmith, I will make the needed adjustments.
